packages/playwright-test/src/reporters/merge.ts

```diff
@@ -114,25 +115,34 @@ async function mergeEvents(dir: string, shardReportFiles: string[], printStatus:
   const configureEvents: JsonEvent[] = [];
   const beginEvents: JsonEvent[] = [];
   const endEvents: JsonEvent[] = [];
-  const shardEvents = await extractAndParseReports(dir, shardReportFiles, printStatus);
-  shardEvents.sort((a, b) => {
-    const shardA = a.metadata.shard?.current ?? 0;
-    const shardB = b.metadata.shard?.current ?? 0;
-    return shardA - shardB;
-  });
-  const allTestIds = new Set<string>();
+
+  const blobs = await extractAndParseReports(dir, shardReportFiles, printStatus);
+  blobs.sort((a, b) => a.file.localeCompare(b.file));
+  const saltSet = new Set<string>();
+
   printStatus(`merging events`);
-  for (const { parsedEvents } of shardEvents) {
+
+  for (const { file, parsedEvents } of blobs) {
+    const metadataEvent = parsedEvents.find(e => e.method === 'onBlobReportMetadata')!;
+
+    // Generate unique salt for each blob.
+    // Prefer metadata.reportName, but fallback to reportFileName because it's unique.
```

Comment:
I guess this only makes sense in case of shards as shard number is not a part of reportName but is included in the file name. In this case maybe just add the shard id to the name?

Reply:
Since `reportName` is optional, we need something stable and unique to use for the salt.